cloud/cluster/cluster.go

```diff
@@ -70,9 +77,16 @@ func (c *Cluster) loop() {
 				sort.Sort(NodeSorter(nodes))
 				outgoing = c.state.setAndDiff(nodes)
 			}
+			elapsed := time.Since(c.priorIterationTime)
+			log.Infof("putting changes on channel, time since last iteration: %s", time.Since(c.priorIterationTime))
+			if c.Stats != nil {
```

Comment:
Only if we end up merging this to master:  I'm not a fan of putting if checks around the stats receiver every time we want to use it, we should initialize it to a nil/fake stats receiver on construction if none is provided instead.

Reply:
agreed